### PR TITLE
Configured profile tomcat.pp and module params.pp

### PR DESCRIPTION
**Pull request by bdeputat**

### CODE REVIEW

**Comment by bdeputat** [comment]
Good day to all. I ask you to see the changes in the manifests ::profile::tomcat.pp and ::tomcat::params.pp  and say your wishes. Thank you!

File: modules/profile/manifests/tomcat.pp [hunk 1]
```diff
@@ -1,26 +1,32 @@
-class profile::tomcat (  
-    $dports         = ['80', '8080', '8080'],
-    $ssh_user       = 'if083',
-    $ssh_group      = 'wheel',
-    $ssh_password   = 'derferterela',
-) {
-
+class profile::tomcat(
+  String $man_user       = 'manager',
```
Comment (ipleten) [inline modules/profile/manifests/tomcat.pp]: Why we have variable type specified here?
But in other place no String List or Integers?
It should be consistent across all classes.

File: modules/tomcat/manifests/params.pp [hunk 2]
```diff
@@ -2,29 +2,31 @@
...
 
-class tomcat::params
-{
+class tomcat::params (
+# config variables for aplication
+$man_user		      = "$profile::tomcat::man_user",	
+$password             = "$profile::tomcat::password", 
+$host_name            = "$profile::tomcat::host_name",
+$docBase              = "$profile::tomcat::docBase", 
+){
```
Comment (ipleten) [inline modules/tomcat/manifests/params.pp]: Why some variables are parameters...

File: modules/tomcat/manifests/params.pp [hunk 3]
```diff
@@ -2,29 +2,31 @@
...
 # config variables for tomcat
-$user             = 'tomcat'
-$password         = 'tomcat' 
-$group            = 'tomcat'
-$java_home        = '/usr/java/jdk1.8.0_162/jre'
-$catalina_home    = '/usr/share/tomcat'
-$catalina_base    = '/var/lib/tomcat/webapps' 
-$confdir          = '/etc/tomcat'
-$packages         = ['tomcat', 'tomcat-webapps', 'tomcat-admin-webapps']
+$user 				  = 'tomcat'
```
Comment (ipleten) [inline modules/tomcat/manifests/params.pp]: but other are hard-coded in module code?
Move them to parameters section as well.

**Review by ipleten — CHANGES_REQUESTED** [review]

File: modules/tomcat/manifests/params.pp (hunk 3 above)
Comment (bdeputat) [reply modules/tomcat/manifests/params.pp]: Variables like $host_name are global and are transmitted from the profile tomcat.pp . Variables like such as $ user = 'tomcat', $ group = 'tomcat', $catalina_home        = '/usr/share/tomcat' - these are default local parameters that do not need to be parameterized.

File: modules/profile/manifests/tomcat.pp (hunk 1 above)
Comment (bdeputat) [reply modules/profile/manifests/tomcat.pp]: Error corrected.

File: modules/tomcat/manifests/params.pp (hunk 2 above)
Comment (bdeputat) [reply modules/tomcat/manifests/params.pp]: Thank you for your comments. Below explanation ...

File: modules/tomcat/manifests/params.pp [hunk 4]
```diff
@@ -3,24 +3,26 @@
 #
 
 class tomcat::params (
-# config variables for aplication
+# config global variables for aplication
 $man_user		      = "$profile::tomcat::man_user",	
 $password             = "$profile::tomcat::password",
```
Comment (ipleten) [inline modules/tomcat/manifests/params.pp]: oh my.... Why do you  have variables from PROFILE like default in MODULE?!

**Comment by bdeputat** [comment]
Good evening all.
Please check the new version of the manifest for the server tomcat
Fixed bugs in the module and profile. The default settings for example.local

**Review by turnopil — COMMENTED** [review]
+  if ($docBase == 'sample') 
+  {
+	$appdir             = "$catalina_home/webapps/$docBase"
+  }
+  else {
+  	$appdir           = "$catalina_home/webapps/$docBase"

Maybe better left only $appdir="$catalina_home/webapps/$docBase" . IF and ELSE do the same things in your code

File: modules/profile/manifests/tomcat.pp [hunk 5]
```diff
@@ -1,36 +1,54 @@
...
+
+# Base variables
+  $ssh_user       		= 'if083'
+  $ssh_group      		= 'wheel'
+  $ssh_password   		= 'tmw'
+  $dports         		= ['80', '8080']
+  
+  class { 'tomcat':
+    tomcat_version    	=> $tomcat_version,
+    dns_name      		=> $dns_name,
```
Comment (ipleten) [inline modules/profile/manifests/tomcat.pp]: indentation.

**Comment by bdeputat** [comment]
turnopil thanks for the comment.
I agree.
Already fixed)

File: modules/tomcat/manifests/init.pp [hunk 6]
```diff
@@ -5,26 +5,111 @@
...
+  
+# Сonfig default variables for tomcat
+  $catalina_home      = '/usr/share/tomcat'
+  $confdir            = "$catalina_home/conf"
+  $tomcat_conf_file   = "$confdir/tomcat.conf"
+  $server_conf_file   = "$confdir/server.xml"
+  $users_conf_file    = "$confdir/tomcat-users.xml"
+
+# # Making sure the  aplication directory is present/created Create
+  if ($docBase == 'sample')
```
Comment (ipleten) [inline modules/tomcat/manifests/init.pp]: I don't understand why you need that if statement?

File: modules/tomcat/manifests/init.pp [hunk 7]
```diff
@@ -5,26 +5,111 @@
...
-     ensure      => running,
+     enable           => true,
+     ensure           => running,
      #hasrestart => true,
      #hasstatus  => true,
-     require    => File["$users_conf_file"],
+     require          => File["$users_conf_file"],
   }
+
   service { 'httpd':
```
Comment (ipleten) [inline modules/tomcat/manifests/init.pp]: You can't manage http service in tomcat module.

File: modules/tomcat/manifests/proxy.pp [hunk 8]
```diff
@@ -5,30 +5,32 @@
...
   # config variables for mod_proxy
-  $host_name            = $tomcat::params::host_name,
-#  $proxyfdir            = $tomcat::params::proxyfdir,
-  $proxy_conf_file      = $tomcat::params::proxy_conf_file,
+#  $dns_name            = $tomcat::params::dns_name,
+  $dns_name,
 )
 {
+  # config variables for mod_proxy
+  $proxy_conf_file      = "/etc/httpd/conf.d/tomcat.conf"
```
Comment (ipleten) [inline modules/tomcat/manifests/proxy.pp]: Also you can manage httpd from tomcat module.
You need to have defined type in httpd module which you will call in  profiles::webapp::tomcat profiles.

File: modules/tomcat/templates/proxy.conf.epp [hunk 9]
```diff
@@ -1,10 +1,10 @@
-<%- | String $host_name
+<%- | String $dns_name
```
Comment (ipleten) [inline modules/tomcat/templates/proxy.conf.epp:1]: All  template should be moved to httpd module

File: modules/tomcat/templates/tomcat.conf.epp [hunk 10]
```diff
@@ -36,7 +36,7 @@ CATALINA_TMPDIR="/var/cache/tomcat/temp"
 
 # You can pass some parameters to java here if you wish to
 #
-JAVA_OPTS="-Xmin<%= $java_ram %> -Xmax<%= $java_ram %>"
+JAVA_OPTS="-Xmin<%= $java_heap %> -Xmax<%= $java_heap %>"
```
Comment (ipleten) [inline modules/tomcat/templates/tomcat.conf.epp]: Are you sure  that java heap memory setting are called Xmin and Xmax but not Xms and Xmx?

**Review by ipleten — CHANGES_REQUESTED** [review]
You can't manage anything related to httpd in tomcat module.

File: modules/profile/manifests/tomcat/tomcat.pp [hunk 11]
```diff
@@ -0,0 +1,59 @@
+class profile::tomcat::tomcat
```
Comment (ipleten) [inline modules/profile/manifests/tomcat/tomcat.pp:1]: profile::tomcat::tomcat? really? 
Why not just profiles::tomcat?

File: modules/profile/manifests/tomcat/tomcat.pp [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    java_home      => $java_home,
+    java_heap      => $java_heap,
+  }
+
+# Configure mod_proxy
+  class { 'profile::tomcat::proxy':
+     dns_name         => $dns_name,
+  }
+
+  base::ssh_user { $ssh_user:
```
Comment (ipleten) [inline modules/profile/manifests/tomcat/tomcat.pp:44]: in basenode?

File: modules/profile/manifests/tomcat/proxy.pp [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    path    => '/usr/bin:/usr/sbin:/bin:/usr/local/bin',
+    require => Package['httpd'],
+  }
+
+# Configured  /etc/httpd/conf.d/tomcat.conf for dns_name and AJP13 protocol
+  $proxy_conf_hash = {
+    'dns_name'        => $dns_name,
+  }
+
+  file { $proxy_conf_file:
```
Comment (ipleten) [inline modules/profile/manifests/tomcat/proxy.pp:25]: that would be great to move to http MODULE, but i'm fine with such solution as well.

**Review by ipleten — CHANGES_REQUESTED** [review]

File: modules/mysql/manifests/install.pp [hunk 14]
```diff
@@ -6,22 +6,11 @@
...
-  ensure   => installed,
-# name     => "mysql${community_major_v}${community_minor_v}-community-release",
-  name     => "mysql-community-server-${community_major_v}.${community_minor_v}.16-1.el${community_minor_v}.x86_64",
-  provider => rpm,
-  source   => $url_repo,
-}
 package { 'mysql-community-server':
   ensure  => present,
-  require => Package['mysql-community-repo'],
+  require => Yumrepo['mysql-repo'],
```
Comment (ipleten) [inline modules/mysql/manifests/install.pp:13]: ?

**Review by ipleten — APPROVED** [review]

File: modules/profile/manifests/tomcat/tomcat.pp (hunk 11 above)
Comment (bdeputat) [reply modules/profile/manifests/tomcat/tomcat.pp:1]: fixed

File: modules/profile/manifests/tomcat/proxy.pp (hunk 13 above)
Comment (bdeputat) [reply modules/profile/manifests/tomcat/proxy.pp:25]: httpd module will use for other people..

File: modules/tomcat/templates/proxy.conf.epp (hunk 9 above)
Comment (bdeputat) [reply modules/tomcat/templates/proxy.conf.epp:1]: moved to web profile

**Review by oleksdiam — APPROVED** [review]

**Pull request merged**